.ci/apm-beats-update.groovy

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env groovy
+@Library('apm@current') _
+
+pipeline {
+  agent { label 'linux && immutable' }
+  environment {
+    REPO = 'apm-server'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    NOTIFY_TO = credentials('notify-to')
+    GITHUB_CHECK_ITS_NAME = 'APM Server Beats update'
+    PATH = "${env.PATH}:${env.WORKSPACE}/bin"
+    HOME = "${env.WORKSPACE}"
+    GOPATH = "${env.WORKSPACE}"
+  }
+  options {
+    timeout(time: 2, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '100', artifactNumToKeepStr: '30', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+    rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
+    quietPeriod(10)
+  }
+  triggers {
+    issueCommentTrigger('(?i).*/run\\s+(?:apm-beats-update\\W+)?.*')
+  }
+  stages {
+    /**
+     Checkout the code and stash it, to use it on other stages.
+    */
+    stage('Checkout') {
+      options { skipDefaultCheckout() }
+      steps {
+        deleteDir()
+        gitCheckout(basedir: "beats")
+        script {
+          dir("beats"){
+            env.GO_VERSION = readFile(".go-version").trim()
+            def regexps =[
+              "^devtools/mage.*",
+              "^libbeat/scripts/Makefile",
+            ]
+            env.BEATS_UPDATED = isGitRegionMatch(patterns: regexps)
+
+            // Skip all the stages except docs for PR's with asciidoc changes only
+            env.ONLY_DOCS = isGitRegionMatch(patterns: [ '.*\\.asciidoc' ], comparator: 'regexp', shouldMatchAll: true)
+          }
+        }
+      }
+    }
+    /**
+    updates beats updates the framework part and go parts of beats.
+    Then build and test.
+    Finally archive the results.
+    */
+    stage('Update Beats') {
+      options { skipDefaultCheckout() }
+      when {
+        beforeAgent true
+        anyOf {
+          branch 'master'
+          branch "\\d+\\.\\d+"
+          branch "v\\d?"
+          tag "v\\d+\\.\\d+\\.\\d+*"
+          allOf {
+            expression { return env.BEATS_UPDATED != "false" || isCommentTrigger() }
+            changeRequest()
+          }
+
+        }
+      }
+      steps {
+        withGithubNotify(context: 'Check Apm Server Beats Update') {
+          deleteDir()
+          gitCheckout(basedir: "${BASE_DIR}",
+            repo: "git@github.com:elastic/${REPO}.git",
+            branch: 'master',
+            credentialsId: 'credentials-id',
+            githubNotifyFirstTimeContributor: false,
+            depth: 1,
+            reference: "/var/lib/jenkins/.git-references/${REPO}.git"
+          )
+          dir("${BASE_DIR}"){
+            sh(label: 'Update Beats script', script: './ci/script/apm-server-update-beats.sh')
+          }
```

Comment:
I do not think this step results in what it should be. When running `./ci/script/apm-server-update-beats.sh` it eventually runs the `make update-beats` cmd from the APM Server. This does not update the beats dependencies from any specific PR though, but from the according branch (`master`, `7.x`, etc.), therefore the changes from the PR would not be considered.

Reply:
yes, the next step it is to make `make update-beats` to accept the source in `${WORKSPACE}/beats` that it is the code from the PR, I've to figure out how to tell it to `govendor`, maybe it is only a matter of put it on the GOPATH

Reply:
we have only to set `BEATS_VERSION` to the commit or the git reference before to call `apm-server-update-beats.sh`

Reply:
with https://github.com/elastic/apm-server/pull/3228 it would use the commit from the PR